doc/source/fq_nmod_sparse_mat.rst

```diff
@@ -0,0 +1,295 @@
+
+.. function:: void fq_nmod_sparse_mat_split_vertical(fq_nmod_sparse_mat_t M1, fq_nmod_sparse_mat_t M2, const fq_nmod_sparse_mat_t B, slong r, const fq_nmod_ctx_t ctx)
+
+    Splits ``B`` vertically into two submatrices ``M1`` and ``M2``, dividing at row ``r``
+
+
+Permutation
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_permute_cols(fq_nmod_sparse_mat_t M, slong *Q, const fq_nmod_ctx_t ctx)
+
+    Permutes the columns indices of ``M`` according to ``Q``, and re-sorts each row
+
+.. function:: void fq_nmod_sparse_mat_permute_rows(fq_nmod_sparse_mat_t M, slong *P, const fq_nmod_ctx_t ctx)
+
+    Permutes the rows of ``M`` according to ``P``
+
+
+Randomization
+--------------------------------------------------------------------------------
+
+
+.. function:: void fq_nmod_sparse_mat_randtest(fq_nmod_sparse_mat_t M, flint_rand_t state, slong min_nnz, slong max_nnz, const fq_nmod_ctx_t ctx)
+
+    Makes ``M`` a sparse matrix with between ``min_nnz`` and ``max_nnz`` nonzero 
+    entries per row, with individual entries generated by fq_nmod_randtest
+
+
+Output
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_print_pretty(const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Prints the matrix ``M`` to ``stdout`` in a human-readable format
+
+
+Comparison
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_is_zero(fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Checks if the given matrix ``M`` is trivial (empty), returning `1` if so and `0` 
+    otherwise
+
+.. function:: void fq_nmod_sparse_mat_equal(const fq_nmod_sparse_mat_t M1, const fq_nmod_sparse_mat_t M2, slong ioff, const fq_nmod_ctx_t ctx)
+
+    Checks if ``M1`` equals ``M2``, returning `1` if so and `0` otherwise
+
+
+Transpose
+--------------------------------------------------------------------------------
+
+.. function:: void fq_sparse_mat_transpose(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Transposes ``M`` into the matrix ``N`` (must have swapped rows and columns)
+
+
+Arithmetic
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_mat_neg(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_ctx_t ctx)
+
+    Sets ``N`` to the negation of ``M``
+
+.. function:: void fq_nmod_sparse_mat_scalar_mul_fq_nmod(fq_nmod_sparse_mat_t N, const fq_nmod_sparse_mat_t M, const fq_nmod_t c, const fq_nmod_ctx_t ctx)
+
+    Sets ``N`` to the scalar multiple of ``M`` by ``c``
+
+.. function:: void fq_nmod_sparse_mat_add(fq_nmod_sparse_mat_t O, const fq_nmod_sparse_mat_t M, const fq_nmod_sparse_mat_t N, const fq_nmod_ctx_t ctx)
+
+    Sets ``O`` to the sum of ``M`` and ``N``
```

Comment:
Can we rename O to something that doesn't look like 0. Also in the actual functions. (passim)